src/index.ts

```diff
@@ -0,0 +1,36 @@
+
+const Credstash = require('@stridespark/credstash');
+const credstash = new Credstash();
+
+export function fetchCred(name: string): Promise<string> {
+    return new Promise<string>(
+        (resolve, reject) =>
+            credstash.get(name, (err: any, secret: string) => {
+                if (err) reject(err);
+                else resolve(secret);
+            })
+    );
+}
+
+export async function fetchAppCred(env: string, appName: string, name: string, defaultValue: string): Promise<string> {
+    try {
+        const appVal = fetchCred(`${env}.${appName}.${name}`)
+        const result = await appVal;
+        return result;
+    } catch (err) {
+        // if a different error, return it
```

Comment:
A little worried about swallowing errors here.  Why should we do it and could we at least log them?